internal/scm/azuredevops/azuredevops.go

```diff
@@ -0,0 +1,540 @@
+	}
+}
+
+func (a *AzureDevOps) getPullRequest(ctx context.Context, branchName string, rid string) (*git.GitPullRequest, error) {
+	status := &git.PullRequestStatusValues.Active
+
+	prs, err := a.gitClient.GetPullRequests(ctx, git.GetPullRequestsArgs{
+		RepositoryId: &rid,
+		SearchCriteria: &git.GitPullRequestSearchCriteria{
+			SourceRefName: &branchName,
+			Status:        status,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(*prs) == 0 {
+		return nil, nil
+	}
+
+	return &(*prs)[0], nil
+}
+
+func pullRequestStatus(pr *git.GitPullRequest) scm.PullRequestStatus {
+	switch {
+	case *pr.MergeStatus == git.PullRequestAsyncStatusValues.Succeeded:
+		return scm.PullRequestStatusSuccess
+	case *pr.Status == git.PullRequestStatusValues.Abandoned || *pr.Status == git.PullRequestStatusValues.Completed:
+		return scm.PullRequestStatusClosed
+	case *pr.MergeStatus == git.PullRequestAsyncStatusValues.Conflicts ||
+		*pr.MergeStatus == git.PullRequestAsyncStatusValues.Failure ||
+		*pr.MergeStatus == git.PullRequestAsyncStatusValues.RejectedByPolicy:
+		return scm.PullRequestStatusError
+	case *pr.Status == git.PullRequestStatusValues.Active:
+		return scm.PullRequestStatusPending
+	default:
+		return scm.PullRequestStatusUnknown
+	}
+}
+
+// GetOpenPullRequest gets a pull request for one specific repository
+func (a *AzureDevOps) GetOpenPullRequest(ctx context.Context, repo scm.Repository, branchName string) (scm.PullRequest, error) {
+	r := repo.(repository)
+
+	prs, err := a.gitClient.GetPullRequests(ctx, git.GetPullRequestsArgs{
+		RepositoryId: &r.rid,
+		SearchCriteria: &git.GitPullRequestSearchCriteria{
+			Status:        &git.PullRequestStatusValues.Active,
+			SourceRefName: &branchName,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(*prs) == 0 {
+		return nil, nil
+	}
+
+	return convertPullRequest(&(*prs)[0]), nil
+}
+
+// MergePullRequest merges a pull request
+func (a *AzureDevOps) MergePullRequest(ctx context.Context, pullReq scm.PullRequest) error {
+	pr := pullReq.(pullRequest)
+
+	_, err := a.gitClient.UpdatePullRequest(ctx, git.UpdatePullRequestArgs{
+		GitPullRequestToUpdate: &git.GitPullRequest{
+			Status: &git.PullRequestStatusValues.Completed,
+			LastMergeSourceCommit: &git.GitCommitRef{
+				CommitId: &pr.lastMergeSourceCommitID,
+			},
+		},
+		RepositoryId:  &pr.repoID,
+		PullRequestId: &pr.id,
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ClosePullRequest closes a pull request
+func (a *AzureDevOps) ClosePullRequest(ctx context.Context, pullReq scm.PullRequest) error {
+	pr := pullReq.(pullRequest)
+
+	_, err := a.gitClient.UpdatePullRequest(ctx, git.UpdatePullRequestArgs{
+		GitPullRequestToUpdate: &git.GitPullRequest{
+			Status: &git.PullRequestStatusValues.Abandoned,
+		},
+		RepositoryId:  &pr.repoID,
+		PullRequestId: &pr.id,
+	})
+	if err != nil {
+		return err
+	}
+
+	newObjectID := "0000000000000000000000000000000000000000"
```

Comment:
Please add a comment of what is happening here

Reply:
Added a comment but let me know if it's not sufficient.  I pulled this value from the docs here: https://learn.microsoft.com/en-us/rest/api/azure/devops/git/refs/update-refs?view=azure-devops-rest-7.1&tabs=HTTP

Reply:
Please add the link to the docs as well and we should be good here